Revit_Core_Engine/Convert/Facade/FromRevit/Opening.cs

```diff
@@ -43,11 +44,76 @@ public static partial class Convert
 
         public static oM.Facade.Elements.Opening FacadeOpeningFromRevit(this FamilyInstance familyInstance, RevitSettings settings = null, Dictionary<string, List<IBHoMObject>> refObjects = null)
         {
-            BH.Engine.Reflection.Compute.RecordError("Conversion of Openings from Revit for the Facade discipline is not yet implemented.");
-            return null;
+            return familyInstance.FacadeOpeningFromRevit(null, settings, refObjects);
+        }
+
+        public static oM.Facade.Elements.Opening FacadeOpeningFromRevit(this FamilyInstance familyInstance, HostObject host = null, RevitSettings settings = null, Dictionary<string, List<IBHoMObject>> refObjects = null)
+        {
+            if (familyInstance == null)
+                return null;
+
+            settings = settings.DefaultIfNull();
+
+            string refId = familyInstance.Id.ReferenceIdentifier(host);
+            oM.Facade.Elements.Opening opening = refObjects.GetValue<oM.Facade.Elements.Opening>(refId);
+            if (opening != null)
+                return opening;
+
+            FrameEdgeProperty frameEdgeProperty = familyInstance.FrameEdgeProperty(settings, refObjects);
+            BH.oM.Geometry.ISurface location = familyInstance.OpeningSurface(host, settings);
+
+            List<FrameEdge> edges = new List<FrameEdge>();
+            if (location == null)
+            {
+                if (host == null)
+                {
+                    BH.Engine.Reflection.Compute.RecordWarning(String.Format("Location of the window could not be retrieved from the model (possibly it has zero area or lies on a non-planar face). A window object without location has been returned. Revit ElementId: {0}", familyInstance.Id.IntegerValue));
```

Comment:
Is it always a window or possibly the warning should be generalised to _opening_?

Reply:
Fixed in latest commit, chaged to opening.